teuthology/misc.py

```diff
@@ -917,7 +917,7 @@ def reboot(node, timeout=300, interval=30):
     reboot_start_time = time.time()
     while time.time() - reboot_start_time < timeout:
         time.sleep(interval)
-        if node.is_online or node.reconnect():
+        if node.is_online(do_test=True) or node.reconnect():
```

Comment:
Not sure that is method should do all this checks. Maybe have `remote.ensure_online` do the next:
```
    def ensure_online(self):
        self.run(args="true")
```
since the _run_ is supposed to call reconnect if socket connection failed,
and this loop will look like:
```
 while time.time() - reboot_start_time < timeout:
        time.sleep(interval)
        try:
            node.ensure_online()
            break
        except:
            log.......
```

Reply:
see #1467